src/lsdb/nested/datasets/generation.py

```diff
@@ -53,22 +182,39 @@ def generate_parquet_file(n_base, n_layer, path, file_per_layer=True, npartition
         The number of rows per n_base row to generate for a nested layer.
         Alternatively, a dictionary of layer label, layer_size pairs may be
         specified to created multiple nested columns with custom sizing.
-    path : str,
-        The path to the parquet file to write to if `file_per_layer` is `False`,
-        and otherwise the path to the directory to write the parquet file for
-        each layer.
-    file_per_layer : bool, default=True
-        TODO: Currently only True is supported.
-        If True, write each layer to its own parquet file. Otherwise, write
-        the generated to a single parquet file representing a nested dataset.
-    npartitions : int, default=1
-        The number of Dask partitions to split the generated data into for each layer.
-    seed : int, default=None
+    seed : int
         A seed to use for random generation of data
+    ra_range : tuple
+        A tuple of the min and max values for the ra column in degrees
+    dec_range : tuple
+        A tuple of the min and max values for the dec column in degrees
+    search_filter : AbstractSearch
+        A search filter to apply to the generated data. Currently supports the
+        ConeSearch and BoxSearch filters. Note that if provided,
+        this will override the `ra_range` and `dec_range` parameters.
+    **kwargs :
+        Additional keyword arguments to pass to `lsdb.from_dataframe`.
 
     Returns
     -------
-    None
+    Catalog
+        The constructed LSDB Catalog.
+
+    Examples
+    --------
+    >>> from lsdb.nested.datasets import generate_catalog # doctest: +SKIP
```

Comment:
But I would love to use `generate_catalog` in other doc-string-test-things!! Is this slow, or just annoying? or non-deterministic with other machines?

Reply:
The doctest skipping is a hold-over from the lsdb.nested port,  but these should be fully usable in doctests. Just be mindful of setting a seed when expecting certain output and being aware of how large of a dataset you're trying to load, I've enabled doctests here, but just assign the results to variables to not check against a certain output.

(I did notice doctest and pylint were butting heads a bit on the string dataframe output, so that may come up as well in future doctests)